documentation/smg/16-serialiser.md

```diff
@@ -0,0 +1,275 @@
+
+__NB.__ For struct arrays `struct2cell` produces a Rank-(N+1) cell array, this means that for struct arrays, the limit on rank is 6.
+
+#### Cell array
+Cell array data (tag 23) are serialised as:
+
+RANK |     tag    | dim1 | dim2 | ... | Data
+-----|------------|------|------|:---:|------
+NULL | 32 + tag   | 0 
+0    |    tag     |      |      |     | Data
+1    | 32 + tag   | nElem|      |     | Data
+2    | 64 + tag   | dim1 | dim2 |     | Data
+N    | N<<5 + tag | dim1 | dim2 | ... | Data
+
+Where `data` is the concatenation of the serialisation of each element.
+
+#### Object Array
+Object array data (tag 23) are serialised as:
+
+RANK |     tag    | dim1 | dim2 | ... |Data|     ...            |    ...     |   ...   |  ...
+-----|------------|------|------|:---:|----|--------------------|------------|---------|------
+NULL | 32 + tag   | 0    |      |     | 33 | length(class_name) | class_name |         |
+0    |    tag     |      |      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+1    | 32 + tag   | nElem|      |     | 33 | length(class_name) | class_name | ser_tag | Data |
+2    | 64 + tag   | dim1 | dim2 |     | 33 | length(class_name) | class_name | ser_tag | Data |
+N    | N<<5 + tag | dim1 | dim2 | ... | 33 | length(class_name) | class_name | ser_tag | Data |
+
+Where
+`33` is a 1-D char tag, 
+`length(class_name)` is stored as a `uint32`, 
+`class_name` is a char string, 
+`ser_tag` is a `uint8` tag (described below) and
+`data` is the serialisation of the object by means described by `ser_tag`
+
+##### ser_tag
+The object array's `ser_tag` contains information about the means by which the object has been serialised.
+It can have one of three values:
+
+0) The object has been serialised by its own `serialize` method
+1) The object has been serialised by the `saveobj` method or function
+2) The object has been serialised by calling `struct` on the object
+
+#### Function handles
+Function handles are serialised differently depending on their type. 
+As function handles in MATLAB are purely scalar objects, the rank-component of the tag is utilised for tagging the type of function handle (thus the method to [de]serialise), these types are:
+
+Handle type         |   tag   | value
+--------------------|---------|-------
+Simple/Class simple | 32 + 25 | 57
+Anonymous           | 64 + 25 | 89
+Scoped              | 96 + 25 | 121
+
+Function handles are serialised as:
+Handle type         |   tag   | data                                       | ... 
+--------------------|---------|--------------------------------------------|-----------------------------------------
+Simple/Class simple |  57     | function name as serialised string         |
+Anonymous           |  89     | anonymous function as serialised string    | relevant workspace as serialised struct
+Scoped              |  121    | function parentage as serialised cell array| 
+
+### Tags
+Tag| Meaning
```

Comment:
For ease of reading the document, it could be good to put this at the start of the new format section. Noting that if the 5 bits of tag are used as above, then this schema is full - it cannot be extended further. It looks quite comprehensive, but just wondering if this is future-proof.